### DISCUSSION TITLE
What is WorkflowSpec.executor.serviceAccountName used for?

### DISCUSSION THREAD

**Discussion by jeffmax** [argoproj/argo-workflows#9872] [Q&A] [answered]
What is [WorkflowSpec.executor.serviceAccountName](https://argoproj.github.io/argo-workflows/fields/#executorconfig) used for? 

How does this compare to just [WorkflowSpec.serviceAccountName](https://argoproj.github.io/argo-workflows/fields/#executorconfig)?

I first encountered this field in this post:[Practical Argo Workflows Hardening](https://blog.argoproj.io/practical-argo-workflows-hardening-dd8429acc1ce) in the example in the "Setting Default Secure Configuration" section.

Thanks!

**Comment by terrytangyuan** [comment, marked as answer]
The executor serviceAccountName will override the workflow level one.

**Reply by jeffmax** [reply]
Thank you!
